platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -372,8 +372,26 @@ export default class HangingProtocolService extends PubSubService {
    * for example, a prior view hanging protocol will NOT show the active study
    * specifically, but will show another study instead.
    */
-  public setActiveStudyUID(activeStudyUID: string): void {
+  public setActiveStudyUID(activeStudyUID: string) {
+    if (!activeStudyUID || activeStudyUID === this.activeStudy?.StudyInstanceUID) {
+      return;
+    }
     this.activeStudy = this.studies.find(it => it.StudyInstanceUID === activeStudyUID);
+    return this.activeStudy;
+  }
+
+  public hasStudyUID(studyUID: string): boolean {
+    return this.studies.some(it => it.StudyInstanceUID === studyUID);
+  }
+
+  public addStudy(study) {
+    if (!this.hasStudyUID(study.StudyInstanceUID)) {
+      this.studies.push(study);
+    }
+  }
+
+  public setDisplaySets(displaySets) {
```

Comment:
I don't understand why we need this, since `.run` already has `displaySets` to work with.

```javascript
public run({ studies, displaySets, activeStudy, activeStudyUID }, protocolId, options = {}) {
```

It also seems like we're mixing things up by adding `hpservice.setDisplaySets`. How about letting `run` handle it?